src/esp/sensor/VisualSensor.cpp

```diff
@@ -3,17 +3,68 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "VisualSensor.h"
+#include <Magnum/EigenIntegration/Integration.h>
 
 #include <utility>
 
 #include "esp/gfx/RenderTarget.h"
 
 namespace esp {
 namespace sensor {
-VisualSensor::VisualSensor(scene::SceneNode& node, SensorSpec::ptr spec)
-    : Sensor{node, std::move(spec)}, tgt_{nullptr} {}
 
-VisualSensor::~VisualSensor() = default;
+VisualSensorSpec::VisualSensorSpec() : SensorSpec() {
+  sensorType = SensorType::Color;
+}
+
+void VisualSensorSpec::sanityCheck() {
+  SensorSpec::sanityCheck();
+  bool isVisualSensor =
+      (sensorType == SensorType::Color || sensorType == SensorType::Depth ||
+       sensorType == SensorType::Normal || sensorType == SensorType::Semantic);
+  CORRADE_ASSERT(
+      isVisualSensor,
+      "VisualSensorSpec::sanityCheck(): sensorType must be Color, Depth, "
+      "Normal, or Semantic", );
+  if (noiseModel == "Gaussian" || noiseModel == "Poisson" ||
+      noiseModel == "SaltAndPepper" || noiseModel == "Speckle") {
+    CORRADE_ASSERT(
+        sensorType == SensorType::Color,
+        "VisualSensorSpec::sanityCheck(): sensorType must be Color if "
+        "noiseModel is Gaussian, Poisson, SaltAndPepper, or Speckle", );
+  }
+  if (noiseModel == "Redwood") {
+    CORRADE_ASSERT(
```

Comment:
Can these noise models even be used from C++?

Reply:
So far I did not see the noise models are used anywhere in the C++, but maybe I am wrong. 

I think we can keep the sanity check like this since it is coded up. It can be easily modified in the future.

But we need to make sure the logic is correct at least. For example, when "redwood" is noise model, the sensor type must be depth sensor.

Reply:
Ok sounds good! I got all the logic for the noiseModels from doing a global search for the noiseModels
<img width="1424" alt="Screen Shot 2021-02-25 at 2 19 53 PM" src="https://user-images.githubusercontent.com/15386768/109212088-b87c5600-7774-11eb-935d-fab41f2aadc9.png">